### PR TITLE
Updates cms-sanity example to use Sanity v3 (#45662

### LINKED ISSUE #46450: SyntaxError in examples cms-sanity
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 13.2.2-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Which example does this report relate to?

cms-sanity

### What browser are you using? (if relevant)

Chrome 110.0.5481.100 (Official Build) (x86_64)

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

https://github.com/vercel/next.js/tree/canary/examples/cms-sanity

I have an error after clicking Deploy button in readme file:
> SyntaxError: Named export 'LIST_NEST_MODE_HTML' not found. The requested module '@portabletext/toolkit' is a CommonJS module, which may not support all module.exports as named exports.

<details>
<summary>Screenshot ✨</summary>
image
</details>

I have the same issue after cloning at localhost:

<details>
<summary>Screenshot ✨</summary>
image
</details>

### Expected Behavior

Successful deployment

### To Reproduce

Go to https://github.com/vercel/next.js/tree/canary/examples/cms-sanity
Click Deploy button in README.md and try to deploy example to vercel

### PR DESCRIPTION
**Pull request by snorrees**
## Bug

- [x] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

Fixes issues reported, for instance here: https://github.com/vercel/community/discussions/1449
(The example does not deploy atm and uses an old version of Sanity).
## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [x] Related issues linked using `fixes #number`
- [ ] [e2e](https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs) tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see [`contributing.md`]

* Updates the Sanity studio to use the latest v3 version. This required changing some files, as per migrating to Studio v3 docs.
* Updated the README to reflect the changed procedure. 
* Update the next app to use the latest version of next-sanity. Changed related code as a result.

### Testing
Tested using this deploy url (uses the branch from the forke repo):
https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fsanity-io%2Fnext.js%2Ftree%2Fsanity-v3%2Fexamples%2Fcms-sanity&repository-name=cms-sanity&project-name=cms-sanity&demo-title=Blog+using+Next.js+%26+Sanity&demo-description=On-demand+ISR%2C+sub-second+as-you-type+previews&demo-url=https%3A%2F%2Fnext-blog-sanity.vercel.app%2F&demo-image=https%3A%2F%2Fuser-images.githubusercontent.com%2F110497645%2F182727236-75c02b1b-faed-4ae2-99ce-baa089f7f363.png&integration-ids=oac_hb2LITYajhRQ0i4QznmKH7gx

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

fixes #46450

### CODE REVIEW

**Review by jankaifer — DISMISSED** [review]
LGTM

File: examples/cms-sanity/components/cover-image.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import cn from 'classnames'
-import Image from 'next/image'
+import Image from 'next/legacy/image'
```
Comment (leerob) [inline examples/cms-sanity/components/cover-image.js:2]: Should we use the new version instead?

**Review by jankaifer — DISMISSED** [review]

**Review by leerob — APPROVED** [review]

File: examples/cms-sanity/lib/sanity.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
 export const urlForImage = (source) =>
   imageBuilder.image(source).auto('format').fit('max')
 
-export const usePreviewSubscription =
-  createPreviewSubscriptionHook(sanityConfig)
+export const usePreview = definePreview({
+  projectId: sanityConfig.projectId,
+  dataset: sanityConfig.dataset,
+  onPublicAccessOnly: () => console.log('lol'),
```
Comment (leerob) [inline examples/cms-sanity/lib/sanity.js:13]: Needed?

**Review by leerob — DISMISSED** [review]

File: examples/cms-sanity/lib/sanity.js (hunk 2 above)
Comment (snorrees) [reply examples/cms-sanity/lib/sanity.js]: Nice catch! (I'm emberrased)

**Review by leerob — DISMISSED** [review]

**Comment by yariknashivan** [comment]
please merge this update, it fixes many issues

**Comment by leerob** [comment]
Looks like there's a merge conflict 🙏

**Review by balazsorban44 — APPROVED** [review]
Thanks!

**Pull request merged**